types.go

```diff
@@ -1781,10 +1783,15 @@ func NewMessage(data []byte) *Message {
 
 // done closes the internal doneSignal channel to let the receiver know that this message has been acted upon
 func (m *Message) done() {
-	// TODO: move initialization in ctor and use ctor everywhere?
-	if m.doneSignal != nil {
-		close(m.doneSignal)
+	if m.doneSignal == nil {
+		return
 	}
+	func(once *sync.Once) {
+		once.Do(func() {
+			close(m.doneSignal)
```

Comment:
Don't we need to set it to `nil` after closing?

Reply:
you can only ever close it once, thanks to the once sync call.
so, it's not needed

Reply:
the check for nil above is because you could call done() on a wrongly instanciated message, and that would panic.